### PR TITLE
Add openstack ssl provider in add_cloud_metadata

### LINKED ISSUE #21392: Beats common "add cloud metadata" function doesn't use HTTPS
To use the  OpenStack API with Auditbeat, HTTPS is required, but the Beats common "add cloud metadata" function doesn't (just uses plain HTTP). 

Currently, there is no workaround to enable the connection to OpenStack API to use HTTPS
- Version: 7.9

### PR DESCRIPTION
**Pull request by jsoriano**
## What does this PR do?

Add a new provider to query metadata from OpenStack deployments using HTTPS.
Add the usual SSL settings available in other features that support TLS.

## Why is it important?

Some OpenStack deployments are configured to use SSL in the metadata endpoints.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] ~~I have made corresponding change to the default configuration files~~
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Related issues

- Fixes #21392.

### CODE REVIEW

File: libbeat/processors/add_cloud_metadata/config.go [hunk 1]
```diff
@@ -20,12 +20,16 @@ package add_cloud_metadata
...
 
 type config struct {
-	Timeout   time.Duration `config:"timeout"`   // Amount of time to wait for responses from the metadata services.
-	Overwrite bool          `config:"overwrite"` // Overwrite if cloud.* fields already exist.
-	Providers providerList  `config:"providers"` // List of providers to probe
+	Timeout   time.Duration     `config:"timeout"`   // Amount of time to wait for responses from the metadata services.
+	TLS       *tlscommon.Config `config:"ssl"`       // TLS configuration
+	Overwrite bool              `config:"overwrite"` // Overwrite if cloud.* fields already exist.
+	Providers providerList      `config:"providers"` // List of providers to probe
+
```
Comment (andrewkroh) [inline libbeat/processors/add_cloud_metadata/config.go:32]: Extra newline.

File: libbeat/processors/add_cloud_metadata/http_fetcher.go [hunk 2]
```diff
@@ -129,16 +130,21 @@ func (f *httpMetadataFetcher) fetchRaw(
...
 	if err != nil {
 		return urls, errors.Wrap(err, "failed to unpack add_cloud_metadata config")
 	}
+	scheme := "http"
+	if config.TLSConfig.IsEnabled() {
+		scheme = "https"
+	}
 	for _, uri := range metadataURIs {
-		urls = append(urls, "http://"+config.MetadataHostAndPort+uri)
+		urls = append(urls, scheme+"://"+config.MetadataHostAndPort+uri)
```
Comment (andrewkroh) [inline libbeat/processors/add_cloud_metadata/http_fetcher.go:147]: By changing the scheme for all providers to HTTPS it means that the processor will really only ever work for openstack-nova when TLS is configured. So if you were deploying in a multiple environments you would need to customize the config for each one, minimizing the effectiveness of the auto-detection.

Perhaps we could register two instances of the openstack-nova provider, one for HTTP and one for HTTPS. `provider` would get an extra flag to indicate https. Then the `add_cloud_metadata` would automatically work any provider (including openstack-nova w/ https enabled if the machine trusted the CA cert). The user's TLS config options would used with any provider connecting over TLS.

Comment (jsoriano) [reply libbeat/processors/add_cloud_metadata/http_fetcher.go:148]: Ok, I have added it as a different provider called `openstack-ssl`/`nova-ssl`. Let me know what you think.

**Review by andrewkroh — APPROVED** [review]
LGTM. Thanks for humoring me with the changes.

**Pull request merged**